test/tests_for_sdkmr/EE2Runjob_test.py

```diff
@@ -0,0 +1,268 @@
+from installed_clients.WorkspaceClient import Workspace
+from installed_clients.CatalogClient import Catalog
+
+
+def test_run_as_admin():
+    """
+    A basic unit test of the run() method with an administrative user.
+
+    This test is a fairly minimal test of the run() method. It does not exercise all the
+    potential code paths or provide all the possible run inputs, such as job parameters, cell
+    metadata, etc.
+    """
+    # The amount of mocking required here implies the method should be broken up into smaller
+    # classes that are individually mockable. Or maybe it's just really complicated and this
+    # is the best we can do. Worth looking into at some point though.
+
+    # We intentionally do not check the logger methods as there are a lot of them and this is
+    # already a very large test. This may be something to be added later when needed.
+    sdkmr = create_autospec(SDKMethodRunner, spec_set=True, instance=True)
+    logger = create_autospec(Logger, spec_set=True, instance=True)
+    ws = create_autospec(Workspace, spec_set=True, instance=True)
+    catutils = create_autospec(CatalogUtils, spec_set=True, instance=True)
+    catalog = create_autospec(Catalog, spec_set=True, instance=True)
+    condor = create_autospec(Condor, spec_set=True, instance=True)
+    kafka = create_autospec(KafkaClient, spec_set=True, instance=True)
+    mongo = create_autospec(MongoUtil, spec_set=True, instance=True)
+    slack = create_autospec(SlackClient, spec_set=True, instance=True)
+    sdkmr.get_logger.return_value = logger
+    sdkmr.get_workspace.return_value = ws
+    sdkmr.get_catalog_utils.return_value = catutils
+    catutils.get_catalog.return_value = catalog
+    sdkmr.get_condor.return_value = condor
+    sdkmr.get_kafka_client.return_value = kafka
+    sdkmr.get_slack_client.return_value = slack
+    sdkmr.get_mongo_util.return_value = mongo
+    sdkmr.get_user_id.return_value = "someuser"
+    sdkmr.get_token.return_value = "tokentokentoken"
+    sdkmr.check_as_admin.return_value = True
+    sdkmr.save_job.return_value = "603051cfaf2e3401b0500982"
+    ws.get_object_info3.return_value = {"paths": [["1/2/3"], ["4/5/6"]]}
+    catalog_resources = {
+        "client_group": "grotesquememlong",
+        "request_cpus": "4",
+        "request_memory": "32",
+    }
+    catutils.get_normalized_resources.return_value = catalog_resources
+    condor.extract_resources.return_value = CondorResources(
+        "4", "2600", "32", "grotesquememlong"
+    )
+    catalog.get_module_version.return_value = {"git_commit_hash": "git5678"}
+    condor.run_job.return_value = SubmissionInfo("cluster42", {}, None)
+    retjob = Job()
+    retjob.id = ObjectId("603051cfaf2e3401b0500982")
+    retjob.status = "created"
+    mongo.get_job.return_value = retjob
+
+    # set up the class to be tested and run the method
+    rj = EE2RunJob(sdkmr)
+    params = {
+        "method": "lolcats.lol_unto_death",
```

Comment:
I hope this is a real kbase app. If not, I propose it be put top of the wishlist for Q2 projects.